cassandra-mesos-scheduler/src/test/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/PersistedCassandraFrameworkConfigurationTest.java

```diff
@@ -49,4 +61,81 @@ public void testGetDefaultRackDc() {
         assertEquals("RAC1", rackDc.getRack());
         assertEquals("DC1", rackDc.getDc());
     }
+
+    @Test
+    public void separationOfFrameworkNameAndClusterNamePreservesOriginalClusterName() throws Exception {
+        final State state = createInitializedState("CassandraFrameworkConfiguration", "/CassandraFrameworkConfiguration_v0.2.0.bin");
+
+        final PersistedCassandraFrameworkConfiguration configuration = new PersistedCassandraFrameworkConfiguration(
+            state,
+            "frameworkName should be cassandra.cluster not this value",
+            15,
+            15,
+            "2.1.4",
+            1.0,
+            2048,
+            2048,
+            1024,
+            1,
+            1,
+            "*",
+            "./backup",
+            ".",
+            true,
+            true,
+            "rack1",
+            "dc1",
+            Collections.<ExternalDc>emptyList(),
+            "clusterName should be cassandra.cluster not this value"
+        );
+
+        assertThat(configuration.frameworkName()).isEqualTo("cassandra.cluster");
+        assertThat(configuration.clusterName()).isEqualTo("cassandra.cluster");
+    }
+
+    @Test
+    public void generatingNewConfigAllowsFrameworkNameAndClusterNameToBeDifferent() throws Exception {
+        final State state = new InMemoryState();
+
+        final PersistedCassandraFrameworkConfiguration configuration = new PersistedCassandraFrameworkConfiguration(
+            state,
+            "cassandra.frameworkName",
+            15,
+            15,
+            "2.1.4",
+            1.0,
+            2048,
+            2048,
+            1024,
+            1,
+            1,
+            "*",
+            "./backup",
+            ".",
+            true,
+            true,
+            "rack1",
+            "dc1",
+            Collections.<ExternalDc>emptyList(),
+            "clusterName"
+        );
+
```

Comment:
This code has nothing to do with the environment variables in `Main`.  This test is specifically there to test and ensure that when initialized that `clusterName` and `frameworkName` can now be two different values since that is the desire from the user.

The test case above this one, reads an existing Persisted Configuration and makes sure that the cluster name stays the same value as was computed before while the proto is "upgraded".